# dynamic-programming-for-bio-informatics

This repository contains an implementation of the Smith-Waterman algorithm for local sequence alignment. The Smith-Waterman algorithm is used in bioinformatics to identify similarities and alignments between biological sequences such as DNA or protein sequences.
Sequence Alignment or sequence comparison lies at heart of the bioinformatics, which describes the way of arrangement of DNA/RNA or protein sequences, in order to identify the regions of similarity among them. It is used to infer structural, functional and evolutionary relationship between the sequences.

## Usage

To use this program, follow the instructions below:

1. Clone the repository to your local machine or download the source code files.

2. Make sure you have Python 3 installed on your machine.

3. Install the required dependencies by running the following command:

    ```
    pip install numpy
    ```

4. Run the program using the following command:

    ```
    python sequence_alignment.py
    ```

5. You will be prompted to choose between manually entering two sequences or generating random sequences. If you choose manual input, enter the sequences when prompted. Otherwise, the program will generate random sequences for you.

6. The program will display the input sequences, perform the Smith-Waterman algorithm for local sequence alignment, and output the aligned sequences along with a visual representation of the optimal alignment path.

## License

Feel free to use, modify, and distribute this code. Contributions are welcome! If you have any suggestions, bug reports, or feature requests, please open an issue or submit a pull request.

## Acknowledgments

This implementation is based on the Smith-Waterman algorithm and is inspired by bioinformatics applications. The algorithm was developed by Temple F. Smith and Michael S. Waterman.
